src/app/commands/cmd_symmetry_mode.cpp

```diff
@@ -22,16 +24,44 @@ class SymmetryModeCommand : public Command {
   Command* clone() const override { return new SymmetryModeCommand(*this); }
 
 protected:
+  void onLoadParams(const Params& params) override;
   bool onEnabled(Context* context) override;
   bool onChecked(Context* context) override;
   void onExecute(Context* context) override;
+  std::string onGetFriendlyName() const override;
+
+private:
+    int m_mode = int(app::gen::SymmetryMode::NONE);
 };
 
 SymmetryModeCommand::SymmetryModeCommand()
   : Command(CommandId::SymmetryMode(), CmdUIOnlyFlag)
 {
 }
 
+std::string SymmetryModeCommand::onGetFriendlyName() const
+{
+  switch (m_mode) {
+    case int(app::gen::SymmetryMode::HORIZONTAL):
+      return Strings::symmetry_toggle_horizontal();
+    case int(app::gen::SymmetryMode::VERTICAL):
+      return Strings::symmetry_toggle_vertical();
+    default:
+      return Strings::symmetry_toggle();
+  }
+}
+
+void SymmetryModeCommand::onLoadParams(const Params& params)
+{
+  Doc* doc = UIContext::instance()->activeDocument();
+  DocumentPreferences& docPref = Preferences::instance().document(doc);
+  m_mode = int(docPref.symmetry.mode());
```

Comment:
The default `m_mode` must be `app::gen::SymmetryMode::NONE`, which actually is (see the `else` case below). So you can remove this line and line 57.